src/odemis/gui/cont/tabs/cryo_chamber_tab.py

```diff
@@ -464,6 +467,11 @@ def _load_project_data(self, evt: wx.Event) -> bool:
         logging.debug(f"{len(proj_data['overviews'])} overviews loaded.")
         logging.debug(f"{len(localization_tab.tab_data_model.streams.value)} streams loaded.")
 
+        # re-subscribe to stream updates
+        # NOTE: this must be called with @call_in_wx_main to force the call to happen after existing stream addition events
+        localization_tab._start_streams_subscriber()
+        correlation_tab.correlation_controller._start_streams_subscriber()
```

Comment:
As these functions must be called delayed, just in this specific context, I'd suggest to change it to explicitly do it here, and *not* in the function definition. It'd be this way:
```suggestion
         wx.CallAfter(localization_tab._start_streams_subscriber)
        wx.CallAfter(correlation_tab.correlation_controller._start_streams_subscriber)
```
=> Then drop the `@call_in_wx_main` decorators on these functions (which is more logical, as the functions don't directly touch the GUI).

Reply:
thanks, didnt know about wx.CallAfter